### PR TITLE
Animation: Timeline Keyboard Controls

### PR DESCRIPTION
**Pull request by carloskelly13**
## Summary

Adds focus and a11y keyboard controls to the start, location, and stop controls of the Timeline Bar.

Kapture 2020-08-18 at 20 36 17

## Relevant Technical Choices

Uses key down hook to add key controls to the timing events.

## To-do

Add tests

## Testing Instructions

---

#3444

### CODE REVIEW

File: assets/src/edit-story/components/keyboard/index.js [hunk 1]
```diff
@@ -69,6 +69,9 @@ function useKeyEffectInternal(
   const batchingCallback = useBatchingCallback(callback, deps || []);
   useEffect(
     () => {
+      if (!refOrNode) {
+        return undefined;
```
Comment (carloskelly13) [inline assets/src/edit-story/components/keyboard/index.js]: In some cases the `refOrNode` is null and has no `current`, this is a guard against that.

**Comment by mariano-formidable** [comment]
@carlos-kelly Is this PR still a `draft`? It's in draft mode.

**Comment by carloskelly13** [comment]
@mariano-formidable The impl is complete. I just wanted to get it up while I write tests.

File: assets/src/edit-story/components/animationTimeline/timingBar.js [hunk 2]
```diff
@@ -140,40 +148,168 @@ export default function TimingBar({
...
   );
 
+  useKeyDownEffect(
+    leftRef,
+    { key: ['left', 'right'] },
+    ({ key }) => {
+      let updatedDuration;
+      let updatedOffset;
+      switch (key) {
+        case 'ArrowRight':
```
Comment (mariano-formidable) [inline assets/src/edit-story/components/animationTimeline/timingBar.js:158]: The code looks great!  But question, how did you know that passing in the `key` of "right" would get translated to "ArrowRight" and same with "left"?  Is this documented somewhere or did you console log `key` and see what values came back?

File: assets/src/edit-story/components/animationTimeline/timingBar.js [hunk 3]
```diff
@@ -79,12 +82,13 @@ export default function TimingBar({
...
+  const leftRef = useRef();
+  const middleRef = useRef();
+  const rightRef = useRef();
   const dragStart = useRef(0);
   const startingDuration = useRef(0);
   const startingOffset = useRef(0);
+  const [mountMovable, setMountMovable] = useState(false);
 
   const [internalDuration, setDuration] = useState(duration);
   const [internalOffset, setOffset] = useState(offset);
```
Comment (mariano-formidable) [inline assets/src/edit-story/components/animationTimeline/timingBar.js]: Nit, that we animation peeps should have probably called out sooner, the terminology the animation logic is using is `duration` and `delay`.  Offset definitely works as a concept of how far along the timeline the block is positioned from 0 (so it makes perfect sense), but if we wanted it to map smoothly to our animation objects without any context switch it should probably be called `delay` 😱 

Not  blocker, just something to consider if it make sense.

**Review by mariano-formidable — APPROVED** [review]
Looks great, only a couple nits/questions to consider, but no blockers!  Works great!!!

File: assets/src/edit-story/components/animationTimeline/timingBar.js (hunk 2 above)
Comment (carloskelly13) [reply assets/src/edit-story/components/animationTimeline/timingBar.js:158]: `ArrowRight` is what the browser gives back. That's the official code. I believe mousetrap shortens it when you provide the key in the configuration array above.

File: assets/src/edit-story/components/animationTimeline/timingBar.js (hunk 3 above)
Comment (carloskelly13) [reply assets/src/edit-story/components/animationTimeline/timingBar.js]: Oh I can rename to delay!

**Pull request merged**